pkg/util/sys/device.go

```diff
@@ -399,17 +409,83 @@ func parseUdevInfo(output string) map[string]string {
 	return result
 }
 
-func isDeviceAvailable(executor exec.Executor, devicePath string) (bool, string, error) {
-	CVInventory, err := inventoryDevice(executor, devicePath)
+func isDeviceAvailable(executor exec.Executor, devicePath string) (available bool, rejectedReasons string, outErr error) {
+	output, err := executor.ExecuteCommandWithOutput("lsblk", devicePath,
+		"--bytes", "--noheadings", "--output", "RM,RO,SIZE,TYPE")
 	if err != nil {
-		return false, "", fmt.Errorf("failed to determine if the device %q is available. %v", devicePath, err)
+		return false, "", errors.Wrapf(err, "failed to determine if device %q is available", devicePath)
+	}
+	items := strings.Fields(output)
+	if len(items) != 4 {
+		return false, "", errors.Errorf(
+			"failed to determine if device %q is available. lsblk output should have 4 fields but has %d: %s", devicePath, len(items), items)
 	}
+	rm := items[0]
+	ro := items[1]
+	sizeStr := items[2]
+	devType := items[3]
 
-	if CVInventory.Available {
-		return true, "", nil
+	rejectedList := []string{}
+
+	// must not be removable
+	if rm != "0" { // 0 means not removable, 1 means removable
+		rejectedList = append(rejectedList, "removable")
+	}
+
+	// must not be read-only
+	if ro != "0" { // 0 means not read-only, 1 means read-only
+		rejectedList = append(rejectedList, "read-only")
+	}
+
+	// must be large enough
+	sizeBytes, err := strconv.Atoi(sizeStr)
+	if err != nil {
+		return false, "", errors.Wrapf(err,
+			"failed to determine if device %q is available. failed to interpret reported size %q", devicePath, sizeStr)
 	}
+	if sizeBytes < 5368709120 /* 5368709120 = 5GB */ {
+		rejectedList = append(rejectedList, "insufficient space (< 5GB)")
+	}
+
+	// must be raw device or partition
+	// rejects devType == "lvm" if is an LVM device
+	if devType != DiskType && devType != PartType {
+		rejectedList = append(rejectedList, fmt.Sprintf(`device type %q is not acceptable; should be raw device ("disk") or partition ("part")`, devType))
```

Comment:
I could look into this. That currently isn't supported, however. I think maybe it would be best to leave that for a future PR targeted to 1.7?

Reply:
I think we do support LVs, but not in this function. This is handled in `CheckIfDeviceAvailable()` which calls `IsLV()` to determine if it should use `isLVAvailable()` instead of `isDeviceAvailable()`